gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraphCollection.java

```diff
@@ -224,15 +229,59 @@ public GraphCollection toGraphCollection() {
+   * @param <G> The graph head type.
+   * @param <V> The vertex type.
+   * @param <E> The edge type.
+   * @param <LG> The type of the graph.
+   * @param <GC> The type of the graph collection.
+   * @return a temporal graph collection with default temporal values
+   * @see TemporalGraphCollectionFactory#fromNonTemporalGraphCollection(BaseGraphCollection)
+   */
+  public static <G extends GraphHead, V extends Vertex, E extends Edge,
+    LG extends BaseGraph<G, V, E, LG, GC>, GC extends BaseGraphCollection<G, V, E, LG, GC>>
+  TemporalGraphCollection fromGraphCollection(GC baseGraphCollection) {
+    return TemporalGradoopConfig.fromGradoopFlinkConfig(baseGraphCollection.getConfig())
+      .getTemporalGraphCollectionFactory().fromNonTemporalGraphCollection(baseGraphCollection);
+  }
+
+  /**
+   * Convenience API function to create a {@link TemporalGraphCollection} from an existing
+   * {@link BaseGraphCollection} with valid times depending on the three given
+   * {@link TimeIntervalExtractor} functions
    *
-   * @param graphCollection the existing graph collection instance
-   * @return a temporal graph colection with default temporal values
+   * @param baseGraphCollection the existing BaseGraphCollection instance
+   * @param graphTimeExtractor  the time extractor function for the Graph Heads in the
+   *                            graph collection instance
+   * @param vertexTimeExtractor the time extractor function for the vertices in the graph
+   *                            collection instance
+   * @param edgeTimeExtractor   the time extractor function for the edges in the graph
+   *                            collection instance
+   * @param <G> The graph head type.
+   * @param <V> The vertex type.
+   * @param <E> The edge type.
+   * @param <LG> The type of the graph.
+   * @param <GC> The type of the graph collection.
+   * @return a temporal graph collection with assigned valid time values
    * @see TemporalGraphCollectionFactory#fromNonTemporalGraphCollection(BaseGraphCollection)
    */
-  public static TemporalGraphCollection fromGraphCollection(GraphCollection graphCollection) {
-    return TemporalGradoopConfig.fromGradoopFlinkConfig(graphCollection.getConfig())
-      .getTemporalGraphCollectionFactory().fromNonTemporalGraphCollection(graphCollection);
+  public static <G extends GraphHead, V extends Vertex, E extends Edge,
+    LG extends BaseGraph<G, V, E, LG, GC>, GC extends BaseGraphCollection<G, V, E, LG, GC>>
+  TemporalGraphCollection fromGraphCollection(GC baseGraphCollection,
+      TimeIntervalExtractor<G> graphTimeExtractor,
+      TimeIntervalExtractor<V> vertexTimeExtractor,
+      TimeIntervalExtractor<E> edgeTimeExtractor) {
+    TemporalGradoopConfig temporalGradoopConfig = TemporalGradoopConfig.
+      fromGradoopFlinkConfig(baseGraphCollection.getConfig());
+    return temporalGradoopConfig.getTemporalGraphCollectionFactory().fromNonTemporalDataSets(
```

Comment:
Line brakes not required here. Even brakes your own code convention since you never added line brakes in a return statement within this pull request.